FNNDSC/ChRIS_ui
# Tests

# Background Information
Tests are setup to ensure the code runs as it is supposed to and new changes do not affect the current workflow. 

# Motivation
This PR contains setup for the testing framework, Cypress, tests for the signup, login and Create New Analysis pages.
![Screenshot from 2022-07-02 08-35-49](https://user-images.githubusercontent.com/67243684/176991525-4a7b43
![Screenshot
![Screenshot from 2022-07-02 08-36-46](https://user-images.githubusercontent.com/67243684/176991629-8f0702c9-5ad7-4dd7-b91e-cab5d10f990b.png)
 from 2022-07-02 08-36-22](https://user-images.githubusercontent.com/67243684/176991573-8704052d-a9c5-4462-8b0e-fa12d52534a2.png)
e3-1d6a-4229-ae6e-92d0ad8924a1.png)


# How to Test
- Make sure you have ChRIS backend running on `http://localhost:8000/api/v1/`
- ChRIS frontend should be running on `http://localhost:3000/`
- Install cypress using `npm install cypress --save-dev`
- Run Cypress using  `npm run cypress:open`
- This command opens up the Cypress UI for interactive testing.
